spring-boot-project/spring-boot-autoconfigure/src/test/java/org/springframework/boot/autoconfigure/pulsar/PulsarPropertiesMapperTests.java

```diff
@@ -81,8 +81,10 @@ void customizeClientBuilderWhenHasNoAuthentication() {
 	@Test
 	void customizeClientBuilderWhenHasAuthentication() throws UnsupportedAuthenticationException {
 		PulsarProperties properties = new PulsarProperties();
-		Map<String, String> params = Map.of("param", "name");
-		String authParamString = "{\"param\":\"name\"}";
+		Map<String, String> params = Map.of("simpleParam", "foo", "complexParam",
```

Comment:
We are not exhausting all [possible replacements characters](https://github.com/spring-projects/spring-boot/pull/40493/files#diff-5bb3152ef3fa91e2d911e7c908848b632a31bfd8f298b76536cbbc56c25a1891R149) **but** are covering the majority. 

@philwebb should we add a bit more coverage here?

Reply:
I'm not Phil, but I think this is fine :)